packages/Button/index.d.ts

```diff
@@ -0,0 +1,37 @@
+export default Button;
+
+declare function Button(props: ButtonProps): JSX.Element;
+interface ButtonProps {
+  /** Descriptive a11y text for assistive technologies. By default, text from children node will be used. */
+  a11yText?: string;
+  /** If click events are allowed to propagate up the DOM tree. */
+  canPropagate?: boolean;
+  /** Body content of the button. */
+  children?: React.ReactNode;
+  /** An icon to be included to the left of children content. */
+  icon?: React.ReactNode;
+  /** If the button is in an "active" or "selected" state. */
+  isActive?: boolean;
+  /** If the button is disabled. */
+  isDisabled?: boolean;
+  /** If the button includes a down arrow to the right of children content. */
+  isDropdown?: boolean;
+  /** If the width of the button should span it's parent container (100%). */
+  isFullWidth?: boolean;
+  /** If the button should render in a pending state (with a spinner icon). */
+  isPending?: boolean;
+  /** If it will be rendered as a <button> element. If false, a <span> will be rendered via an accessible <RawButton>. */
+  isSemantic?: boolean;
+  /** If the type attribute should "submit", instead of the default "button". */
+  isSubmit?: boolean;
+  /** The visual style of the button. */
+  kind?: Kinds.ALL;
```

Comment:
no for this PR but, after we merge the constant package we should be able to provide a better descriptive way for this kind of prop.